view/Requirements.php

```diff
@@ -1439,16 +1464,24 @@ function() use ($fileList, $minify, $type) {
 				}
 			);
 
-		// Since url won't be automatically suffixed, add it in here
-		if($this->getSuffixRequirements()) {
-			$q = stripos($combinedURL, '?') === false ? '?' : '&';
-			$combinedURL .= "{$q}m={$entropy}";
-		}
-
 		return $combinedURL;
 	}
 
 	/**
+	 * Given a filename and list of files, generate a new filename unique to these files
+	 *
+	 * @param string $name
+	 * @param array $files
+	 * @return string
+	 */
+	protected function hashedCombinedFilename($combinedFile, $fileList) {
+		$name = pathinfo($combinedFile, PATHINFO_FILENAME);
+		$hash = $this->hashOfFiles($fileList);
+		$extension = File::get_file_extension($combinedFile);
+		return $name . '-' . substr($hash, 0, 7) . '.' . $extension;
```

Comment:
Using a hash also solves the problem of a file being duplicate included if the mtime differs, but the content is the same (since cms js loading treats the different urls as different files, even though only the mtime differs).

It also provides a solution to determining when a file MUST be regenerated, since it's not immediately obvious by looking at the filename whether it is out of date (if the mtime is unreliable).